code-challenges/401challenges/src/main/java/challenges/graph/Graph.java

```diff
@@ -0,0 +1,47 @@
+package challenges.graph;
+
+
+import java.util.HashSet;
+
+public class Graph<T> {
+    private HashSet<Node<T>> vertices;
+
+    public HashSet<Node<T>> getVertices() {
+        return vertices;
+    }
+
+    public void setVertices(HashSet<Node<T>> vertices) {
+        this.vertices = vertices;
+    }
+
+    public Graph(){
+        this.vertices = new HashSet<>();
+    }
+
+    public Node<T>addNode(T data){
+        Node node = new Node(data);
+        this.vertices.add(node);
+        return node;
+    }
+
+    public void addEdge(Node<T> node, Node<T> target){
```

Comment:
I'm not sure what this method does. It checks to see if `node` and `target` are in `vertices`, and if they are, it adds them to `vertices`?

I'm not seeing where the edge is created between the two nodes.